internal/credentials/credentials.go

```diff
@@ -67,14 +67,19 @@ type kubernetesDatabase struct {
 // carries out the important task of indexing Credentials stored in Kubernetes
 // Secrets by repository type + URL.
 func NewKubernetesDatabase(
+	sharedCredentialsNamespaces []string,
 	argoCDNamespace string,
```

Comment:
Since there are multiple bits of configuration to include now, I would have a preference for starting to pass this config as an instance of a new type -- let's say `KubernetesDatabaseConfig`.

Something like this:

```golang
type KubernetesDBConfig struct {
	ArgoCDCredentialBorrowingEnabled bool     `envconfig:"ARGOCD_ENABLE_CREDENTIAL_BORROWING" default:"false"`
	ArgoCDNamespace                  string   `envconfig:"ARGOCD_NAMESPACE" default:"argocd"`
	SharedCredentialsNamespaces      []string `envconfig:"SHARED_CREDENTIALS_NAMESPACES" default:""`
}

func KubernetesDBConfigFromEnv() KubernetesDBConfig {
	cfg := KubernetesDBConfig{}
	envconfig.MustProcess("", &cfg)
	return cfg
}
```

Reply:
Also note that envconfig can parse comma-separated strings into a `[]string` for you. :smile: